sound/soc/sof/core.c

```diff
@@ -234,6 +235,17 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	if (ret < 0)
 		goto fw_trace_err;
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST)
+	/*
+	 * Now register client devices. Only one for now.
+	 * This can fail but errors cannot be propagated.
+	 */
+	ret = sof_client_dev_register(sdev, "sof-ipc-test");
+	if (ret)
+		dev_err(sdev->dev,
+			"error: failed to register client device: %s\n",
+			"sof-ipc_test");
```

Comment:
how is this better than just "error: failed to register \"sof-ipc_test\" client device"? In fact in such cases I find printing the error code useful...